include/boost/redis/impl/ping_request_utils.ipp

```diff
@@ -0,0 +1,33 @@
+/* Copyright (c) 2018-2024 Marcelo Zimbres Silva (mzimbres@gmail.com)
+ *
+ * Distributed under the Boost Software License, Version 1.0. (See
+ * accompanying file LICENSE.txt)
+ */
+
+#include <boost/redis/config.hpp>
+#include <boost/redis/detail/ping_request_utils.hpp>
+#include <boost/redis/request.hpp>
+
+namespace boost::redis::detail {
+
+void compose_ping_request(const config& cfg, request& to)
+{
+   to.clear();
+   to.push("PING", cfg.health_check_id);
+}
+
+system::error_code check_ping_response(system::error_code io_ec, const generic_response& resp)
+{
+   if (io_ec)
+      return io_ec;
+
+   // TODO: we should probably introduce a better error code/logging
+   // for PONG errors. We defer this until
+   // https://github.com/boostorg/redis/issues/104
+   if (resp.has_error())
+      return error::pong_timeout;
```

Comment:
AFAICS if `resp` has an error it will be a RESP3 error which should not be converted into a `pong_timeout`. The latter should occur only if `cancel_after` fires. An error in this `resp` object should be returned to the user when the `headl_check` operation completes and its [diagnostic](https://github.com/boostorg/redis/blob/f955dc01d2c5b9695f347992c0136321a8ff5276/include/boost/redis/adapter/result.hpp#L26) should be logged.